vcd/structure.go

```diff
@@ -94,6 +104,14 @@ func convertStringsToTypeSet(param []string) *schema.Set {
 	return set
 }
 
+// addrOf is a generic function to return the address of a variable
+// Note. It is mainly meant for converting literal values to pointers (e.g. `addrOf(true)`) or cases
+// for converting variables coming out straight from Terraform schema (e.g.
+// `addrOf(d.Get("name").(string))`).
+func addrOf[T any](variable T) *T {
+	return &variable
+}
```

Comment:
Maybe it's just me, but would this be more readable?
```suggestion
func addrOf(variable any) *any {
	return &variable
}
```

Reply:
These are not the same - one is generic (Go compiler creates functions for specific types at build time behind the scenes) and returns precise types (the ones it got via `T`)

In your case, it will return `*interface{}` (`any` is an alias of empty interface `interface{}` https://github.com/golang/go/issues/33232).

Here is an example: https://go.dev/play/p/vUKUOwWeiwn

Reply:
Oh, interesting, thanks for the explanation, thought under the hood they're still the same